src/App.js

```diff
@@ -1,14 +1,22 @@
 import { Route, Routes, Navigate } from "react-router-dom";
 import "./App.css";
 import Mockman from "mockman-js";
-import { Home, Trending } from "./pages";
+import { Home, Trending, Login, Signup } from "./pages";
+import { ToasterWrapper } from "./utils";
+import { useAuth } from "./context/AuthContext";
 
 function App() {
+  const {
+    state: { isLoggedIn },
+  } = useAuth();
   return (
     <div className="App">
+      <ToasterWrapper />
       <Routes>
         <Route path="/" element={<Home />} />
         <Route path="/trending" element={<Trending />} />
+        {!isLoggedIn && <Route path="/signup" element={<Signup />} />}

```

Comment:
U can wrap these inside private route instead doing conditional check on each route

Reply:
login & signup are not private routes. It's a reverse implementation of the private route. if a user is logged in, then he/she can't able to visit the login/signup page. he/she has to log out first to access the login page.